volmdlr/faces.py

```diff
@@ -2710,19 +2720,200 @@ def point2d_with_dimension_to_3d(self, point2d, points_x, points_y, xmin, xmax,
+        points = []
+        for key in self._grids2d:
+            [points_x, points_y, xmin, xmax, ymin, ymax] = key
+
+        for p in points_dim:
+            points.append(self.point2d_with_dimension_to_parametric_frame(p, points_x, points_y, xmin, xmax, ymin, ymax))
+
+        bsplinecurve2d = volmdlr.edges.BSplineCurve2D(bsplinecurve2d.degree, points,
+                                                      bsplinecurve2d.knot_multiplicities,
+                                                      bsplinecurve2d.knots,
+                                                      bsplinecurve2d.weights,
+                                                      bsplinecurve2d.periodic)
+        return bsplinecurve2d
+
+
+    def bsplinecurve2d_with_dimension_to_3d(self, bsplinecurve2d):
+        '''
+        compute the bsplinecurve3d, on a Bspline surface, of a bsplinecurve2d defined in the dimensioned frame
+        '''
+
+        bsplinecurve2d_01 = self.bsplinecurve2d_with_dimension_to_parametric_frame(bsplinecurve2d)
+        bsplinecurve3d = self.bsplinecurve2d_to_3d(bsplinecurve2d_01)
+
+        return bsplinecurve3d
+
+
+    def arc2d_parametric_to_dimension(self, arc2d, points_x, points_y):
+        '''
+        convert a arc2d from the parametric to the dimensioned frame
+        '''
+
+        xmin, xmax, ymin, ymax = 0, 1, 0, 1
+
+        start = self.point2d_parametric_to_dimension(arc2d.start, points_x, points_y, xmin, xmax, ymin, ymax)
+        end = self.point2d_parametric_to_dimension(arc2d.end, points_x, points_y, xmin, xmax, ymin, ymax)
+        interior = self.point2d_parametric_to_dimension(arc2d.interior, points_x, points_y, xmin, xmax, ymin, ymax)
+
+        arc2d_with_dimension = volmdlr.edges.Arc2D(start, interior, end)
+
+        return arc2d_with_dimension
+
+
+    def arc3d_to_2d_with_dimension(self, arc3d, points_x, points_y):
+        '''
+        compute the arc2d of a arc3d, on a Bspline surface, in the dimensioned frame
+        '''
+
+        arc2d = self.arc3d_to_2d(arc3d) #it's a bsplinecurve2d
+        arc2d_with_dimension = self.bsplinecurve2d_parametric_to_dimension(arc2d, points_x, points_y)
+
+        return arc2d_with_dimension #it's a bsplinecurve2d-dimension
+
+
+    def arc2d_with_dimension_to_parametric_frame(self, arc2d):
+        '''
+        convert a arc2d from the dimensioned to the parametric frame
+        '''
+
+        for key in self._grids2d:
+            [points_x, points_y, xmin, xmax, ymin, ymax] = key
+
+        start = self.point2d_with_dimension_to_parametric_frame(arc2d.start, points_x, points_y, xmin, xmax, ymin, ymax)
+        interior = self.point2d_with_dimension_to_parametric_frame(arc2d.interior, points_x, points_y, xmin, xmax, ymin, ymax)
+        end = self.point2d_with_dimension_to_parametric_frame(arc2d.end, points_x, points_y, xmin, xmax, ymin, ymax)
+
+        arc2d = volmdlr.edges.Arc2D(start, interior, end)
+
+        return arc2d
```

Comment:
The other way around now : are you sure an Arc2D in dimensional coordinates is also an Arc2D in parametric coordinates ?